### PR TITLE
chore: data model style updates

### PR DESCRIPTION
**Pull request by vasilev-alex**

### CODE REVIEW

File: packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts [hunk 1]
```diff
@@ -146,36 +139,45 @@ export abstract class BaseSchemaFormatter {
...
         ? Object.fromEntries(
           Object.entries(contextProps).map(([key, value]) => [toSnakeCase(key), value])
         )
         : contextProps),
+
+      [this.options.snakeCase ? 'pre_aggregations' : 'preAggregations']: new ValueWithComments(
+        null,
+        [
+          'Pre-aggregation definitions go here',
+          'Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started',
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts:178]:
```suggestion
          'Pre-aggregation definitions go here.',
          'Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started',
```

File: packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts [hunk 2]
```diff
@@ -146,36 +139,45 @@ export abstract class BaseSchemaFormatter {
...
+          },
+        }),
+        
       ...(this.options.snakeCase
         ? Object.fromEntries(
           Object.entries(contextProps).map(([key, value]) => [toSnakeCase(key), value])
         )
         : contextProps),
+
+      [this.options.snakeCase ? 'pre_aggregations' : 'preAggregations']: new ValueWithComments(
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts:174]: @vasilev-alex Do we need `this.options.snakeCase` anymore? Can't we go with snake case at all times?

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 3]
```diff
@@ -108,18 +108,21 @@ describe('ScaffoldingTemplate', () => {
...
   
-  preAggregations: {
-    // Pre-aggregation definitions go here
-    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
-  },
-  
   joins: {
     Customers: {
       sql: \`\${CUBE}."customerId" = \${Customers}.id\`,
       relationship: \`belongsTo\`
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:114]:
```suggestion
      relationship: \`many_to_one\`
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 4]
```diff
@@ -108,18 +108,21 @@ describe('ScaffoldingTemplate', () => {
...
     }
   },
   
+  dimensions: {
+    id: {
+      sql: \`id\`,
+      type: \`number\`,
+      primaryKey: true
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:122]:
```suggestion
      primary_key: true
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 5]
```diff
@@ -146,29 +146,13 @@ describe('ScaffoldingTemplate', () => {
           content: `cube(\`Customers\`, {
   sql: \`SELECT * FROM public.customers\`,
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:147]: `sql_table` as well

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 6]
```diff
@@ -146,29 +146,13 @@ describe('ScaffoldingTemplate', () => {
...
   
-  preAggregations: {
-    // Pre-aggregation definitions go here
-    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
-  },
-  
   joins: {
     Accounts: {
       sql: \`\${CUBE}."accountId" = \${Accounts}.id\`,
       relationship: \`belongsTo\`
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:152]:
```suggestion
      relationship: \`many_to_one\`
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 7]
```diff
@@ -180,6 +164,22 @@ describe('ScaffoldingTemplate', () => {
...
       type: \`string\`
     }
+  },
+  
+  measures: {
+    count: {
+      type: \`count\`
+    },
+    
+    visitCount: {
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:174]:
```suggestion
    visit_count: {
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 8]
```diff
@@ -180,6 +164,22 @@ describe('ScaffoldingTemplate', () => {
...
+      sql: \`visit_count\`,
+      type: \`sum\`
+    }
+  },
+  
+  preAggregations: {
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:180]:
```suggestion
  pre_aggregations: {
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 9]
```diff
@@ -180,6 +164,22 @@ describe('ScaffoldingTemplate', () => {
...
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:182]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 10]
```diff
@@ -189,26 +189,10 @@ describe('ScaffoldingTemplate', () => {
           content: `cube(\`Accounts\`, {
   sql: \`SELECT * FROM public.accounts\`,
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:190]: `sql_table`

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 11]
```diff
@@ -189,26 +189,10 @@ describe('ScaffoldingTemplate', () => {
...
-    
-    failurecount: {
-      sql: \`\${CUBE}."failureCount"\`,
-      type: \`sum\`
-    }
-  },
-  
   dimensions: {
     id: {
       sql: \`id\`,
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:198]: 2 lines below, `primary_key`

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 12]
```diff
@@ -146,29 +146,13 @@ describe('ScaffoldingTemplate', () => {
...
-    
-    visitCount: {
-      sql: \`visit_count\`,
-      type: \`sum\`
-    }
-  },
-  
   dimensions: {
     id: {
       sql: \`id\`,
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:158]: 2 lines below, `primary_key`

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 13]
```diff
@@ -225,6 +209,22 @@ describe('ScaffoldingTemplate', () => {
...
+    
+    failurecount: {
+      sql: \`\${CUBE}."failureCount"\`,
+      type: \`sum\`
+    }
+  },
+  
+  preAggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:227]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts (hunk 10 above)
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:190]:
```suggestion
  sqlTable: \`public.accounts\`,
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 14]
```diff
@@ -272,12 +275,9 @@ describe('ScaffoldingTemplate', () => {
...
   
-  dimensions: {
-    id: {
-      sql: \`id\`,
-      type: \`number\`,
-      primary_key: true
-    }
+  pre_aggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:280]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 15]
```diff
@@ -366,6 +350,22 @@ describe('ScaffoldingTemplate', () => {
...
+    
+    failurecount: {
+      sql: \`\${CUBE}."failureCount"\`,
+      type: \`sum\`
+    }
+  },
+  
+  pre_aggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:368]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 16]
```diff
@@ -429,17 +437,9 @@ describe('ScaffoldingTemplate', () => {
...
-      primary_key: true
-    },
-    
-    somedimension: {
-      sql: \`\${CUBE}.\\\`someDimension\\\`\`,
-      type: \`string\`
-    }
+  pre_aggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:442]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 17]
```diff
@@ -505,6 +494,17 @@ describe('ScaffoldingTemplate', () => {
...
+  
+  measures: {
+    count: {
+      type: \`count\`
+    }
+  },
+  
+  pre_aggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:507]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 18]
```diff
@@ -515,11 +515,6 @@ describe('ScaffoldingTemplate', () => {
     it('should add options if passed', () => {
       const schemaContext = {
         dataSource: 'testDataSource',
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:517]:
```suggestion
        data_source: 'testDataSource',
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 19]
```diff
@@ -584,6 +567,17 @@ describe('ScaffoldingTemplate', () => {
...
+  
+  measures: {
+    count: {
+      type: \`count\`
+    }
+  },
+  
+  pre_aggregations: {
+    // Pre-aggregation definitions go here
+    // Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:580]:
```suggestion
  pre_aggregations: {
    // Pre-aggregation definitions go here.
    // Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 20]
```diff
@@ -629,11 +625,9 @@ describe('ScaffoldingTemplate', () => {
...
         type: sum
 
-    dimensions:
-      - name: id
-        sql: id
-        type: number
-        primary_key: true
+    pre_aggregations:
+      # Pre-aggregation definitions go here
+      # Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:630]:
```suggestion
    pre_aggregations:
      # Pre-aggregation definitions go here.
      # Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 21]
```diff
@@ -670,6 +652,18 @@ describe('ScaffoldingTemplate', () => {
...
+      - name: count
+        type: count
+
+      - name: visit_count
+        sql: visit_count
+        type: sum
+
+    pre_aggregations:
+      # Pre-aggregation definitions go here
+      # Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:665]:
```suggestion
    pre_aggregations:
      # Pre-aggregation definitions go here.
      # Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 22]
```diff
@@ -706,6 +688,18 @@ describe('ScaffoldingTemplate', () => {
...
+      - name: count
+        type: count
+
+      - name: failurecount
+        sql: "{CUBE}.\\"failureCount\\""
+        type: sum
+
+    pre_aggregations:
+      # Pre-aggregation definitions go here
+      # Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:701]:
```suggestion
    pre_aggregations:
      # Pre-aggregation definitions go here
      # Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts [hunk 23]
```diff
@@ -760,6 +742,18 @@ describe('ScaffoldingTemplate', () => {
...
+      - name: count
+        type: count
+
+      - name: failurecount
+        sql: "{CUBE}.\`failureCount\`"
+        type: sum
+
+    pre_aggregations:
+      # Pre-aggregation definitions go here
+      # Learn more here: https://cube.dev/docs/caching/pre-aggregations/getting-started
```
Comment (igorlukanin) [inline packages/cubejs-schema-compiler/test/unit/scaffolding-template.test.ts:755]:
```suggestion
    pre_aggregations:
      # Pre-aggregation definitions go here.
      # Learn more in the documentation: https://cube.dev/docs/caching/pre-aggregations/getting-started
```

File: packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts (hunk 2 above)
Comment (vasilev-alex) [reply packages/cubejs-schema-compiler/src/scaffolding/formatters/BaseSchemaFormatter.ts:174]: have to keep it for backwards compatibility

**Pull request merged**